controllers/front/paymentreturn.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+* Team Paylike
+*
+*  @author    Team Paylike
+*  @copyright Team Paylike
+*  @license   MIT license: https://opensource.org/licenses/MIT
+*/
+
+class PaylikePaymentReturnModuleFrontController extends ModuleFrontController
+{
+	public function __construct()
+	{
+		parent::__construct();
+		$this->display_column_right = false;
+		$this->display_column_left = false;
+		$this->context = Context::getContext();
+	}
+
+	public function init()
+	{
+		parent::init();
+		$cart = $this->context->cart;
+		if ($cart->id_customer == 0 || $cart->id_address_delivery == 0 || $cart->id_address_invoice == 0 || !$this->module->active)
+			Tools::redirect('index.php?controller=order&step=1');
+
+		$authorized = false;
+		foreach (Module::getPaymentModules() as $module)
+			if ($module['name'] == 'paylike')
+			{
+				$authorized = true;
+				break;
+			}
+
+		if (!$authorized)
+			die($this->module->l('Paylike payment method is not available.', 'paymentreturn'));
+
+		$customer = new Customer($cart->id_customer);
+		if (!Validate::isLoadedObject($customer))
+			Tools::redirect('index.php?controller=order&step=1');
+
+		$paylike = new Paylike();
+		$total = $cart->getOrderTotal(true, Cart::BOTH);
+		$currency = new Currency((int)$cart->id_currency);
+		$paylikeapi = new PaylikeAPI(Configuration::get('PAYLIKE_SECRET_KEY'));
+		$amount = Tools::ps_round($total, 2) * 100;
+		$status_paid = Configuration::get('PS_OS_PAYMENT');
+		$status_error = Configuration::get('PS_OS_ERROR');
+		$transactionid = Tools::getValue('transactionid');
+		$params = array(
+			'paylike_redirect' => 1,
+			'transactionid' => $transactionid,
+			'amount' => $amount
+		);
+
+		$transaction_failed = false;
+		if (Configuration::get('PAYLIKE_CHECKOUT_MODE') == 'delayed')
+		{
+			$fetch = $paylikeapi->transactions->fetch($transactionid);
+			if ($fetch && $fetch->transaction->currency == $currency->iso_code && $fetch->transaction->custom->cartId == $cart->id && $fetch->transaction->amount == $amount)
+			{
+				$message =
+				'Trx ID: '.$transactionid.'
+				Authorized Amount: '.($fetch->transaction->amount / 100).'
+				Captured Amount: '.($fetch->transaction->capturedAmount / 100).'
+				Order time: '.$fetch->transaction->created.'
+				Currency code: '.$fetch->transaction->currency;
+
+				if ($paylike->validateOrder((int)$cart->id, $status_paid, $total, $paylike->displayName, $message, array(), null, false, $customer->secure_key))
```

Comment:
If this fails, you should void the transaction.

Actually, you should not void the transaction if the verification of the transaction fails (line 78), as that means someone has been tampering the data, and thus you cannot rely on the transaction ID anyway.

In both scenarios though I think you should display the error page to the user.